tools/federation/src/test/java/demos/FedXWithRemoteRepositoryManager.java

```diff
@@ -0,0 +1,47 @@
+package demos;
+
+import org.eclipse.rdf4j.federated.FedXFactory;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.vocabulary.FOAF;
+import org.eclipse.rdf4j.model.vocabulary.RDF;
+import org.eclipse.rdf4j.repository.Repository;
+import org.eclipse.rdf4j.repository.RepositoryConnection;
+import org.eclipse.rdf4j.repository.RepositoryResult;
+import org.eclipse.rdf4j.repository.manager.RemoteRepositoryManager;
+import org.eclipse.rdf4j.repository.manager.RepositoryManager;
+
+/**
+ * Demo which shows an example of a local federation which uses repositories from a remote RDF4J instance as federation
+ * members.
+ * 
+ * @author Andreas Schwarte
+ *
+ */
+public class FedXWithRemoteRepositoryManager {
+
+	public static void main(String[] args) throws Exception {
+
+		// connection URL of a RDF4J server which manages the repositories
+		String serverUrl = "http://localhost:8080/rdf4j-server";
+		RepositoryManager repoManager = new RemoteRepositoryManager(serverUrl);
+
+		// assumes that remote repositories exist
+		Repository repo = FedXFactory.newFederation()
+				.withRepositoryResolver(repoManager)
+				.withResolvableEndpoint("my-repository-1")
+				.withResolvableEndpoint("my-repository-2")
+				.create();
+
+		repo.init();
```

Comment:
This reminds me: calling `init()` should not be mandatory. I'm not sure if the FedXRepository implements that (as it's a fairly new thing).

Reply:
Good point, just tested and it does not yet work. I'll track an issue and provide a fix for 3.1